src/fides/api/ctl/sql_models.py

```diff
@@ -514,7 +517,50 @@ class CustomFieldDefinition(Base):
     )
     active = Column(BOOLEAN, nullable=False, default=True)
 
-    UniqueConstraint("name", "resource_type")
+    @classmethod
+    def create(
+        cls: Type[PrivacyDeclaration],
+        db: Session,
+        *,
+        data: dict[str, Any],
+        check_name: bool = False,  # this is the reason for the override
+    ) -> PrivacyDeclaration:
+        """
+        Overrides base create to avoid unique check on `name` column
+        and to cleanly handle uniqueness constraint on name/resource_type
+        """
+        try:
+            return super().create(db=db, data=data, check_name=check_name)
+        except IntegrityError as e:
+            if cls.name_resource_index in str(e):
+                raise KeyOrNameAlreadyExists(
+                    "Custom field definitions must have unique names for a given resource type"
+                )
+            raise e
+
+    def update(self, db: Session, *, data: Dict[str, Any]) -> FidesBase:
+        """Overrides base update to cleanly handle uniqueness constraint on name/resource type"""
+        try:
+            return super().update(db=db, data=data)
+        except IntegrityError as e:
+            if CustomFieldDefinition.name_resource_index in str(e):
+                raise KeyOrNameAlreadyExists(
+                    "Custom field definitions must have unique names for a given resource type"
+                )
+            raise e
+
+    # unique index on the lowername/resource type for case-insensitive name checking per resource type
+    name_resource_index = (
```

Comment:
just storing the string as a constant variable here to allow it to be referenced when looking for the index name in error messages above.